frontend/talentsearch/src/js/components/pool/PoolAdvertisementPage.tsx

```diff
@@ -149,38 +150,10 @@ const PoolAdvertisement = ({
     poolAdvertisement.nonessentialSkills,
   );
 
-  const [, executeMutation] = useCreateApplicationMutation();
-  const handleCreateApplication = () =>
-    executeMutation({ userId, poolId: poolAdvertisement.id })
-      .then((result) => {
-        if (result.data?.createApplication) {
-          navigate(paths.reviewApplication(result.data.createApplication.id));
-          toast.success(
-            intl.formatMessage({
-              defaultMessage: "Application created",
-              id: "U/ji+A",
-              description: "Application created successfully",
-            }),
-          );
-          return result.data?.createApplication.id;
-        }
-        return result.data?.createApplication;
-      })
-      .catch((result) => {
-        toast.error(
-          intl.formatMessage({
-            defaultMessage: "Error application creation failed",
-            id: "tlAiJm",
-            description: "Application creation failed",
-          }),
-        );
-        return Promise.reject(result.error);
-      });
-
-  const applyBtn = hasUserApplied ? (
+  const applyBtn = hasApplied ? (
```

Comment:
There's a bit of a gap here though I'm not sure if it's worth worrying about.  This decision is made when the page renders but they could render this page then apply in a different tab.  If they come back here and click apply again it will overwrite their application!  It would be better if this decision were made when they actually clicked the button, I think.

https://user-images.githubusercontent.com/8978655/198090513-bca8c27a-97b9-407e-9d16-2ceeeee49e7b.mp4

Reply:
I actually didn't check beyond seeing the toast.  If it's actually overwriting the application could you throw a new bug on the board?  Our API probably shouldn't allow this at all.

Reply:
I think you are right, we are only checking if user has applied on the frontend so it is easy to overwrite it with this trick. I'll create a ticket

Reply:
Confirmed it is an issue, created ticket. Great catch! 👏 

https://github.com/GCTC-NTGC/gc-digital-talent/issues/4463